React/Views/RCTSwitch.m

```diff
@@ -11,6 +11,14 @@
 
 @implementation RCTSwitch
 
+// [macOS
+- (void)setOn:(BOOL)on
+{
+  _wasOn = on;
+  [super setOn:on];
+}
+// macOS]
```

Comment:
We could probably move this to the shim layer implementation of `RCTUISwitch` found in `RCTUIKit.m`. But also, should move to a separate PR. 

Some context: Once we started re-implementing these components for Fabric, we realized we didn't properly "shim" (that is, provide an RCTUI* abstraction for macOS) all of the paper components, so we started doing so for Fabric. Fixing bugs at the shim layer (`RCTUIKit`) instead of the paper implementation (`RCTSwitch`) should mean that we get them for free for the fabric implementation (`RCTSwitchComponentView.m`)

Reply:
I'll revert this since you have fixed it with #1871